src/main/java/spoon/support/sniper/internal/ModificationStatus.java

```diff
@@ -0,0 +1,32 @@
+/**
+ * SPDX-License-Identifier: (MIT OR CECILL-C)
+ *
+ * Copyright (C) 2006-2019 INRIA and contributors
+ *
+ * Spoon is available either under the terms of the MIT License (see LICENSE-MIT.txt) of the Cecill-C License (see LICENSE-CECILL-C.txt). You as the user are entitled to choose the terms under which to adopt Spoon.
+ */
+package spoon.support.sniper.internal;
+
+/** represents whether a {@link spoon.support.sniper.internal.SourceFragment} has been modified, and should be reprinted as is or with the normal pretty-printer */
+public enum ModificationStatus {
+	UNKNOWN,
+	MODIFIED,
+	NOT_MODIFIED;
+
+	public static ModificationStatus fromBoolean(Boolean b) {
+		if (b) {
+			return MODIFIED;
+		}
+		return NOT_MODIFIED;
+	}
+
+	public boolean toBoolean() {
+		if (this == MODIFIED) {
+			return true;
+		}
+		if (this == NOT_MODIFIED) {
+			return false;
+		}
+		throw new IllegalStateException();
```

Comment:
shouldn't this be abstract and all Enum constants implement this? Otherwise every new enum item would need a change in this method.

Reply:
that's not necessary to over-engineer, this enum is very unlikely to change